KubeArmor/core/kubeUpdate.go

```diff
@@ -1249,18 +1294,35 @@ func (dm *KubeArmorDaemon) CreateSecurityPolicy(policyType string, securityPolic
 		excludedNamespaces := make(map[string]bool)
 
 		for _, matchExpression := range secPolicy.Spec.Selector.MatchExpressions {
-			if matchExpression.Key == "namespace" {
-				if matchExpression.Operator == "In" {
+			if matchExpression.Key == NamespaceKey {
+				if matchExpression.Operator == InOperator {
 					hasInOperator = true
 					secPolicy.Spec.Selector.NamespaceList = append(secPolicy.Spec.Selector.NamespaceList, matchExpression.Values...)
-				} else if matchExpression.Operator == "NotIn" && !hasInOperator {
+				} else if matchExpression.Operator == NotInOperator && !hasInOperator {
 					for _, value := range matchExpression.Values {
 						excludedNamespaces[value] = true
 					}
 				}
+			} else if matchExpression.Key == LabelKey {
+				if matchExpression.Operator == InOperator {
+					for _, label := range matchExpression.Values {
+						secPolicy.Spec.Selector.MatchExpIdentities = append(secPolicy.Spec.Selector.MatchExpIdentities, label)
+					}
+				} else if matchExpression.Operator == NotInOperator {
+					for _, label := range matchExpression.Values {
+						secPolicy.Spec.Selector.NonIdentities = append(secPolicy.Spec.Selector.NonIdentities, label)
+					}
+				}
```

Comment:
for CSP `In` and `NotIn` operators are being handled differently than KSP any specific reason? iinr in case of ksp `NotIn` rules are being added only if `In` rules are not present which is not the case here with csp.

Reply:
consequently this csp policy will target a workload even though both labels are not present. btw logically it matches NotIn rule.
```yaml
apiVersion: security.kubearmor.com/v1
kind: KubeArmorClusterPolicy
metadata:
  name: csp-match-expr
spec:
  process:
    matchPaths:
      - path: /usr/bin/sleep
      - path: /bin/sleep
  action: Block    
  selector:
    matchExpressions:
      - key: label
        operator: In
        values:
        - not-exists
      - key: label
        operator: NotIn
        values:
        - also-not-exists
```

Reply:
got it will update for csp

Reply:
updated

Reply:
we need to document this as well as when both `matchLabel` and `matchExpression` present both rules should be matched to select a workload.

Reply:
I've mentioned it here - https://github.com/kubearmor/KubeArmor/pull/1999/files#diff-e871007b073b341af07a21f370a61f1338dd6805746955d03cc674e5bb7acd51R173